lib/ui/platform_dispatcher.dart

```diff
@@ -347,7 +356,13 @@ class PlatformDispatcher {
   // ignore: unused_field, field will be used when platforms other than web use these focus APIs.
   Zone _onViewFocusChangeZone = Zone.root;
   set onViewFocusChange(ViewFocusChangeCallback? callback) {
-    _onViewFocusChange = callback;
+    _onViewFocusChange = (ViewFocusEvent event) {
+      _focusedViewId = switch (event.state) {
+        ViewFocusState.focused => event.viewId,
+        ViewFocusState.unfocused => -1,
+      };
+      callback?.call(event);
```

Comment:
The dartdocs should say that the callback is invoked _after_ `focusedViewId` is updated.